### DISCUSSION TITLE
The "Robot System" Parameter on "from Plane" and "get Plane" should be a required input and warn if nothing is connected

### DISCUSSION THREAD

**Discussion by robin-gdwl** [visose/Robots#248] [General]
The **From Plane** and **Get Plane** components do different conversions depending on the robot system specified. 
If no Robot System is connected, the conversion is done to Quaternions. 
No error or warning is shown when no robot system is connected. 
This makes it possible to disconnect the Robot system component during refactoring or on accident and never noticing, that the conversion that is being done is completely different now. 

This is error-prone and difficult to trouble-shoot. 
A Warning if no Robot system is connected would be great.

**Comment by visose** [comment]
Thanks for the suggestion. A warning makes sense for the GetPlane component.

The primary use of FromPlane is really with the robotsystem input disconnected, to get the convention used in the xml files to edit them. So it might be a bit weird to get a warning in the most common case.

Another less error prone option might be to make the robotsystem input a variable input that you can add/remove (like in the create target component), and if added but disconnected it doesn't return any value until you connect a robot.

**Comment by visose** [comment]
How about adding a "remark"? You get a balloon icon but doesn't turn the component orange:
<img width="534" alt="image" src="https://github.com/visose/Robots/assets/5129614/ffd5882a-9b3d-4276-acaa-f873c213c018">

**Reply by robin-gdwl** [reply]
This would definitely help. However, it would not be noticeable when refactoring larger grasshopper files. 
I understand the hesitation to add an orange warning. There are many gh components that work differently when no input is present.
A remark may be a good compromise to alert users to the status without adding warnings to definitions that don't need the warning.

**Reply by visose** [reply]
Yes, Grasshopper in general is not great at alerting on these type of user mistakes. Besides this happening in other components as you say, it does other similar things, like implicit conversions between data types (plane to point, vector, etc.) without showing any warnings, when these often are accidental. Not sure if it's worth making this plugin's components more sensitive to these possible mistakes than what you would expect from normal Grasshopper components/parameters.

**Comment by visose** [comment]
Closing this as stale. The warning/remark behavior for Get Plane / From Plane is a design suggestion rather than an active support question. Please open a fresh feature discussion or PR if this should be revisited.
